libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java

```diff
@@ -744,12 +745,30 @@ public void appendMediaFile(final MediaFile mediaFile, final String mediaUrl, Im
             public void run() {
                 if (URLUtil.isNetworkUrl(mediaUrl)) {
                     String mediaId = mediaFile.getMediaId();
-                    mWebView.execJavaScriptFromString("ZSSEditor.insertImage('" + mediaUrl + "', '" + mediaId + "');");
+                    if (mediaFile.isVideo()) {
+                        String videoPressId = ShortcodeUtils.getVideoPressIdFromShortCode(
+                                mediaFile.getVideoPressShortCode());
+
+                        mWebView.execJavaScriptFromString("ZSSEditor.insertVideo('" + mediaUrl + "', '" +
```

Comment:
I wonder if the `mediaUrl` could be abused here (with infos coming from the server) and if we should escape it (same for the `ZSSEditor.insertImage`).